libs/wingc/src/lsp/rename_request.rs

```diff
@@ -419,4 +439,217 @@ test "pow" {
 		"#,
 		"t1"
 	);
+
+	test_rename_request!(
+		rename_class_property,
+		r#"
+class Cat {
+  pub name: str;
+    //---^
+  new(name: str) {
+    this.name = name;
+       //----
+  }
+
+  inflight pub jump(height: num) {}
+  pub sleep(d: duration) {
+    log("{this.name} sleeps");
+             //----
+  }
+}
+
+let a = new Cat("A") as "a cat";
+let b = new Cat("B") as "b cat";
+
+a.sleep(18h);
+b.sleep(24h);
+log(b.name);
+    //----
+
+test "a cat scares b cat" {
+  a.jump(5);
+  b.jump(-5);
+}
+		"#,
+		"t1"
+	);
+	test_rename_request!(
+		rename_class_property_from_usage,
+		r#"
+class Cat {
+  pub name: str;
+    //----
+  new(name: str) {
+    this.name = name;
+       //----
+  }
+
+  inflight pub jump(height: num) {}
+  pub sleep(d: duration) {
+    log("{this.name} sleeps");
+             //----
+  }
+}
+
+let a = new Cat("A") as "a cat";
+let b = new Cat("B") as "b cat";
+
+a.sleep(18h);
+b.sleep(24h);
+log(b.name);
+    //---^
+
+test "a cat scares b cat" {
+  a.jump(5);
+  b.jump(-5);
+}
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_class_inflight_method,
+		r#"
+class Cat {
+  pub name: str;
+  new(name: str) {
+    this.name = name;
+  }
+
+  inflight pub jump(height: num) {}
+             //---^
+  pub sleep(d: duration) {
+    log("{this.name} sleeps");
+  }
+}
+
+let a = new Cat("A") as "a cat";
+let b = new Cat("B") as "b cat";
+
+a.sleep(18h);
+b.sleep(24h);
+log(b.name);
+
+test "a cat scares b cat" {
+  a.jump(5);
+  //----
+  b.jump(-5);
+  //----
+}
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_class_inflight_method_from_usage,
+		r#"
+class Cat {
+  pub name: str;
+  new(name: str) {
+    this.name = name;
+  }
+
+  inflight pub jump(height: num) {}
+             //----
+  pub sleep(d: duration) {
+    log("{this.name} sleeps");
+  }
+}
+
+let a = new Cat("A") as "a cat";
+let b = new Cat("B") as "b cat";
+
+a.sleep(18h);
+b.sleep(24h);
+log(b.name);
+
+test "a cat scares b cat" {
+  a.jump(5);
+  //----
+  b.jump(-5);
+  //---^
+}
+		"#,
+		"t1"
+	);
+	test_rename_request!(
+		rename_class_preflight_method,
+		r#"
+class Cat {
+  pub name: str;
+  new(name: str) {
+    this.name = name;
+  }
+
+  inflight pub jump(height: num) {}
+  pub sleep(d: duration) {
+    //----^
+    log("{this.name} sleeps");
+  }
+}
```

Comment:
If this class were public, renaming the method would cause errors in any other files that use it. I suspect this will be surprising for most people using the rename functionality.

Until we implement multi-file renames, maybe would should prevent renames of _public_ classes/structs? I'm not sure if it's better to give the user the option to do it anyway and let them fix it since they'll still get the compiler error.

Reply:
opened #6694